cmd/repo_link.go

```diff
@@ -0,0 +1,238 @@
+			cli.StringFlag{Name: "auto-deploy", Usage: "Enable auto deploy of application after each branch change"},
+			cli.StringFlag{Name: "deploy-review-apps", Usage: "Enable auto deploy of review app when new pull request is opened"},
+			cli.StringFlag{Name: "destroy-on-close", Usage: "Auto destroy review apps when pull request is closed"},
+			cli.StringFlag{Name: "hours-before-destroy-on-close", Usage: "Time delay before auto destroying a review app when pull request is closed"},
+			cli.StringFlag{Name: "destroy-on-stale", Usage: "Auto destroy review apps when no deploy/commits has happened"},
+			cli.StringFlag{Name: "hours-before-destroy-on-stale", Usage: "Time delay before auto destroying a review app when no deploy/commits has happened"},
+		},
+		Usage: "Create a repo link between your scm integration and your app",
+		Description: ` Create a repo link between your scm integration and your application:
+	$ scalingo --app my-app repo-link-create <integration-name> <repo-url> [options]
+									   OR
+	$ scalingo --app my-app repo-link-create <integration-uuid> <repo-url> [options]
+
+	List of available integrations:
+	- github => GitHub.com
+	- github-enterprise => GitHub Enterprise (private instance)
+	- gitlab => GitLab.com
+	- gitlab-self-hosted => GitLab Self-hosted (private instance)
+
+	Examples:
+	$ scalingo -a test-app repo-link-create gitlab https://gitlab.com/gitlab-org/gitlab-ce
+	$ scalingo -a test-app repo-link-create github-enterprise https://ghe.example.org/test/frontend-app --branch master --auto-deploy true
+
+		# See also 'repo-link', 'repo-link-update', 'repo-link-delete', 'repo-link-manual-deploy' and 'repo-link-manual-review-app'`,
+		Action: func(c *cli.Context) {
+			if len(c.Args()) != 2 {
+				cli.ShowCommandHelp(c, "repo-link-create")
+				return
+			}
+
+			currentApp := appdetect.CurrentApp(c)
+			integrationType := c.Args()[0]
+			integrationURL := c.Args()[1]
+			branch := c.String("branch")
+			autoDeploy := c.Bool("auto-deploy")
+			deployReviewApps := c.Bool("deploy-review-apps")
+			destroyOnClose := c.Bool("destroy-on-close")
+			hoursBeforeDestroyOnClose := c.Uint("hours-before-destroy-on-close")
+			destroyOnStale := c.Bool("destroy-on-stale")
+			hoursBeforeDestroyOnStale := c.Uint("hours-before-destroy-on-stale")
+
+			params := scalingo.SCMRepoLinkParams{
+				Branch:                   &branch,
+				AutoDeployEnabled:        &autoDeploy,
+				DeployReviewAppsEnabled:  &deployReviewApps,
+				DestroyOnCloseEnabled:    &destroyOnClose,
+				HoursBeforeDeleteOnClose: &hoursBeforeDestroyOnClose,
+				DestroyStaleEnabled:      &destroyOnStale,
+				HoursBeforeDeleteStale:   &hoursBeforeDestroyOnStale,
+			}
+
+			err := repolink.Create(currentApp, integrationType, integrationURL, params)
+			if err != nil {
+				errorQuit(err)
+			}
+		},
+		BashComplete: func(c *cli.Context) {
+			_ = autocomplete.CmdFlagsAutoComplete(c, "repo-link-create")
+		},
+	}
+
+	repoLinkUpdateCommand = cli.Command{
```

Comment:
Add `--no-auto-deploy` to disable. The `CheckAndFillParams function is wrong. There is an issue if the flag is not mentioned.